src/app/api/configuration/configuartionSerice.ts

```diff
@@ -0,0 +1,45 @@
+import prisma from "db";
+import {
+  WeightConfig,
+  WeightConfigDto,
+  weightValueMap,
+} from "./weightConfig.dto";
+
+export async function getTeamWeightConfigs(
+  teamId: string,
+): Promise<WeightConfigDto[]> {
+  const teamWeightConfigs = await prisma.teamWeightConfig.findMany({
+    where: { team_id: teamId },
+  });
+  return teamWeightConfigs.map((config) => {
+    return WeightConfigDto.fromJSON(config);
+  });
+}
+
+export async function saveTeamWeightConfig(
+  teamId: string,
+  weightConfig: WeightConfig,
+): Promise<boolean> {
+  for (const key in weightConfig) {
+    deleteIgnoreErrors(teamId, key);
+    await prisma.teamWeightConfig.create({
+      data: {
+        team_id: teamId,
+        type: key,
+        value: weightValueMap[weightConfig[key].value],
+      },
+    });
+  }
```

Comment:
why don't we fetch all teamId weights, and update the ones that already exists/create the ones that doesn't? 
Not sure I understand the function name "deleteIgnoreErrors" as well, shouldn't it be just deleteTeamWeights?

Reply:
Its more resource optimise to delete everything and re-insert it than to fetch and then update